network_importer/adapters/netbox_api/models.py

```diff
@@ -354,15 +362,21 @@ def delete(self) -> Optional["DiffSyncModel"]:
             NetboxIPAddress: DiffSync object
         """
         if self.device_name:
-            dev = self.diffsync.get(self.diffsync.device, identifier=self.device_name)
-            if dev.primary_ip == self.address:
-                LOGGER.warning(
-                    "Unable to delete IP Address %s on %s, because it's currently the management IP address",
-                    self.address,
-                    self.device_name,
+            try:
+                dev = self.diffsync.get(self.diffsync.device, identifier=self.device_name)
+                if dev.primary_ip == self.address:
+                    LOGGER.warning(
+                        "Unable to delete IP Address %s on %s, because it's currently the management IP address",
+                        self.address,
+                        self.device_name,
+                    )
+                    return None
+
+            except ObjectNotFound:
+                LOGGER.error(
+                    "Unable to delete IP Address %s on %s, because device is not present.", self.name, self.device_name,
```

Comment:
This might be hiding a bigger problem, do you know why the device is not present anymore ? that shouldn't happen as far as I know